src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
-        this.state.value = typeof value === 'string' ? Number.parseInt(value!, 10) : value;
-    }
-
-    /**
-     * Clear the state of this control.
+     * Example utterance: "Set my age to six"
+     *
+     * @param input - Input
      */
-    clear() {
-        this.state = new NumberControlState();
-    }
-
-    private canHandleForEmptyStateValue(input: ControlInput): boolean {
-        try {
-            okIf(this.state.value === undefined);
-
-            if (InputUtil.isIntent(input, GeneralControlIntent.name)) {
-                const { action, target } = unpackGeneralControlIntent(
-                    (input.request as IntentRequest).intent,
-                );
-                okIf(
-                    InputUtil.actionIsMatchOrUndefined(action, [
-                        ...this.props.interactionModel.actions.set,
-                        ...this.props.interactionModel.actions.change,
-                    ]),
-                );
-                okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
-
-                this.handleFunc = this.handleLastQuestionEmptyAndValueNotExisting;
-                return true;
-            } else if (InputUtil.isValueControlIntent(input, AmazonBuiltInSlotType.NUMBER)) {
-                const { action, target } = unpackValueControlIntent((input.request as IntentRequest).intent);
-                okIf(
-                    InputUtil.actionIsMatchOrUndefined(action, [
-                        ...this.props.interactionModel.actions.set,
-                        ...this.props.interactionModel.actions.change,
-                    ]),
-                );
-                okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
-                this.handleFunc = this.handleLastQuestionEmptyAndValueExisting;
-                return true;
-            }
-            return false;
-        } catch (e) {
-            return falseIfGuardFailed(e);
-        }
-    }
-
-    private isScreenEvent(input: ControlInput): boolean {
+    private isSetWithValue(input: ControlInput): boolean {
         try {
-            okIf(InputUtil.isAPLUserEventWithMatchingControlIdAndArgLength(input, this.id, 2));
-            this.handleFunc = this.handleScreenEvent;
+            okIf(InputUtil.isValueControlIntent(input, AmazonBuiltInSlotType.NUMBER));
+            const { action, target, feedback, values } = unpackValueControlIntent(
+                (input.request as IntentRequest).intent,
+            );
+            const valueStr = values[0].slotValue;
+            okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
+            okIf(InputUtil.valueStrDefined(valueStr));
+            okIf(InputUtil.feedbackIsMatchOrUndefined(feedback, FEEDBACK_TYPES));
+            okIf(InputUtil.actionIsMatch(action, this.props.interactionModel.actions.set));
+            this.handleFunc = this.handleSetWithValue;
```

Comment:
here and elsewhere: this is spurious now I think. Pls remove.

Reply:
Removed stale `this.handleFunc` assigns everywhere